stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/options/message/internal/MessageOptionsView.kt

```diff
@@ -156,27 +167,38 @@ internal class MessageOptionsView : FrameLayout {
         val muteEnabled: Boolean,
         val blockEnabled: Boolean,
         val retryMessageEnabled: Boolean,
+        val ownCapabilities: Set<String>,
     ) : Serializable {
         internal companion object {
             operator fun invoke(
                 viewStyle: MessageListViewStyle,
                 channelConfig: Config,
                 hasTextToCopy: Boolean,
                 suppressThreads: Boolean,
-            ) = Configuration(
-                replyEnabled = viewStyle.replyEnabled,
-                threadsEnabled = if (suppressThreads) false else viewStyle.threadsEnabled && channelConfig.isThreadEnabled,
-                editMessageEnabled = viewStyle.editMessageEnabled,
-                deleteMessageEnabled = viewStyle.deleteMessageEnabled,
-                copyTextEnabled = viewStyle.copyTextEnabled && hasTextToCopy,
-                deleteConfirmationEnabled = viewStyle.deleteConfirmationEnabled,
-                reactionsEnabled = viewStyle.reactionsEnabled && channelConfig.isReactionsEnabled,
-                flagEnabled = viewStyle.flagEnabled,
-                pinMessageEnabled = viewStyle.pinMessageEnabled,
-                muteEnabled = viewStyle.muteEnabled,
-                blockEnabled = viewStyle.blockEnabled,
-                retryMessageEnabled = viewStyle.retryMessageEnabled,
-            )
+                ownCapabilities: Set<String>,
+            ): Configuration {
+                val userCanReply = ownCapabilities.contains(ChannelCapabilities.SEND_REPLY)
+                val userCanPinMessage = ownCapabilities.contains(ChannelCapabilities.PIN_MESSAGE)
+                val userCanBlockMembers = ownCapabilities.contains(ChannelCapabilities.BAN_CHANNEL_MEMBERS)
+
+                return Configuration(
+                    replyEnabled = viewStyle.replyEnabled && userCanReply,
```

Comment:
We have a situation here where some of these logic conditions are further defined in `MessageOptionsView.configure()`.

This is not ideal and I would prefer to have all of these sit in a single place, however the crux of the matter is that we do not have certain information such as the ownership of the message at this point, while we do have it in `MessageOptionsView.configure()`. Conversely, it is important to define if reply is enabled here.